crates/circuit/src/slice.rs

```diff
@@ -372,4 +382,37 @@ mod test {
             );
         }
     }
+
+    /// Test SequenceIndex::from_int correctly handles positive and negative indices
+    #[test]
+    fn convert_py_idx() {
+        let cases: IntoIter<(isize, usize, usize), 3> = [
+            (2, 5, 2), // (index, sequence length, expected result)
+            (-2, 5, 3),
+            (0, 2, 0),
+        ]
+        .into_iter();
```

Comment:
You don't need to manually call `into_iter` - `for` loops do it for you.  I think you could drop the type annotation and the `into_iter()` call, and it'd all work the way you want.

If Rust struggles with determining the integer types, you can make the first tuple super explicit as `(2isize, 5usize, 2usize)` and then it should be happy.

Reply:
Great, thanks for the hint! Done in [f9c114c](https://github.com/Qiskit/qiskit/pull/12716/commits/f9c114cfaefbeccdfc3d9fda5af4a2c4b1849afe).